fluster/utils.py

```diff
@@ -73,26 +73,28 @@ def run_pipe_command_with_std_output(
     verbose: bool = False,
     check: bool = True,
     timeout: Optional[int] = None,
-) -> Tuple[str, str]:
+) -> List[str]:
     """Runs a command and returns std output trace"""
-    serr = subprocess.DEVNULL if not verbose else subprocess.PIPE
+    serr = subprocess.DEVNULL if not verbose else subprocess.STDOUT
     if verbose:
         print(f'\nRunning command "{" ".join(command)}"')
 
     try:
-        with subprocess.Popen(
-            command, stdout=subprocess.PIPE, stderr=serr, universal_newlines=True
-        ) as pipe:
-            data = pipe.communicate(timeout=timeout)
-            if check and pipe.returncode:
-                if verbose:
-                    for line in filter(None, data):
-                        print(line, end="")
-                raise subprocess.CalledProcessError(
-                    cmd=command, returncode=pipe.returncode
-                )
-            return data
+        data = subprocess.check_output(
+            command, stderr=serr, timeout=timeout, universal_newlines=True
+        )
+        return data.splitlines()
     except (subprocess.CalledProcessError, subprocess.TimeoutExpired) as ex:
+        odata: List[str] = []
+        if verbose or check:
+            odata = ex.output.decode("utf-8").splitlines()
```

Comment:
KernelCI folks reported that on CalledProcessError, ex.output follows universal_newlines=True, but in my testing with TimeoutExpired, it was left as binary. I need to test this further, it could also be an ABI break in Python that depends on the version.

Reply:
Fixed by detecting the exception type, a bit unfortunate, but works.